File: app/src/main/java/it/anyplace/syncbrowser/FragmentIntentIntegrator.java
package it.anyplace.syncbrowser;
import android.content.Intent;
import android.support.v4.app.Fragment;

import com.google.zxing.integration.android.IntentIntegrator;

// https://stackoverflow.com/a/22320076/1837158
public final class FragmentIntentIntegrator extends IntentIntegrator {

    private final Fragment fragment;

    public FragmentIntentIntegrator(Fragment fragment) {
        super(fragment.getActivity());
        this.fragment = fragment;
    }

    @Override
    protected void startActivityForResult(Intent intent, int code) {
        fragment.startActivityForResult(intent, code);
    }
}

File: app/src/main/kotlin/net/syncthing/lite/fragments/FragmentIntentIntegrator.kt
package net.syncthing.lite.fragments

import android.content.Intent
import android.support.v4.app.Fragment

import com.google.zxing.integration.android.IntentIntegrator

// https://stackoverflow.com/a/22320076/1837158
class FragmentIntentIntegrator(private val fragment: Fragment) : IntentIntegrator(fragment.activity) {

    override fun startActivityForResult(intent: Intent, code: Int) {
        fragment.startActivityForResult(intent, code)
    }
}
